lib/presentation/widget/biometric_authentication/biometric_authentication_viewmodel.dart

```diff
@@ -35,91 +35,70 @@ import 'package:linshare_flutter_app/presentation/localizations/app_localization
 import 'package:linshare_flutter_app/presentation/redux/actions/biometric_authentication_action.dart';
 import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
 import 'package:linshare_flutter_app/presentation/util/router/app_navigation.dart';
+import 'package:linshare_flutter_app/presentation/util/router/route_paths.dart';
 import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
 import 'package:redux/redux.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/list_biometric_kind_extension.dart';
 
 class BiometricAuthenticationViewModel extends BaseViewModel {
   final AppNavigation _appNavigation;
   final AuthenticationBiometricInteractor _authenticationBiometricInteractor;
-  final EnableBiometricInteractor _enableBiometricInteractor;
+  final DisableBiometricInteractor _disableBiometricInteractor;
+  final DeletePermanentTokenInteractor deletePermanentTokenInteractor;
   final GetAvailableBiometricInteractor _getAvailableBiometricInteractor;
-  final GetBiometricSettingInteractor _getBiometricSettingInteractor;
 
   BiometricAuthenticationViewModel(
     Store<AppState> store,
     this._appNavigation,
     this._authenticationBiometricInteractor,
-    this._enableBiometricInteractor,
     this._getAvailableBiometricInteractor,
-    this._getBiometricSettingInteractor
-    ) : super(store);
+    this.deletePermanentTokenInteractor,
+    this._disableBiometricInteractor
+  ) : super(store);
 
-  void getBiometricSetting() {
+  void gotoSignIn() {
     store.dispatch((Store<AppState> store) async {
```

Comment:
I think you are using dispatching wrongly, you can only dispatch Actions referenced in actions files or ThunkActions so they can be caught by reducers, if you plan to not use reducers you don't have to dispatch an action